avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordClassGenerator.java

```diff
@@ -627,140 +627,162 @@ private String replaceSingleDollarSignWithDouble(String str) {
   private void populateBuilderClassBuilder(TypeSpec.Builder recordBuilder, AvroRecordSchema recordSchema,
       SpecificRecordGenerationConfig config) throws ClassNotFoundException {
     boolean canThrowMissingFieldException = false;
+    // Split Builder constructors from other Builder and other record if # of fields is > 50
+    boolean splitMethods = recordSchema.getFields().size() > 50;
+    int chunkSize = 25;
     recordBuilder.superclass(ClassName.get(CompatibleSpecificRecordBuilderBase.class));
     CodeBlock.Builder otherBuilderConstructorFromRecordBlockBuilder = CodeBlock.builder();
     CodeBlock.Builder otherBuilderConstructorFromOtherBuilderBlockBuilder = CodeBlock.builder();
+    CodeBlock.Builder buildMethodChunkBuilder = CodeBlock.builder();
     CodeBlock.Builder buildMethodCodeBlockBuilder = CodeBlock.builder()
         .beginControlFlow("try")
         .addStatement("$1L record = new $1L()", recordSchema.getName().getSimpleName());
 
     List<MethodSpec> accessorMethodSpecs = new ArrayList<>();
-    int fieldIndex = 0;
+    int fieldIndex = 0, chunkCounter = 0;
```

Comment:
nit - i know this is valid syntax, i'd just prefer 2 different declaration lines

Reply:
c/c++ like this way :)